folium/utilities.py

```diff
@@ -345,3 +345,42 @@ def png_pack(png_tag, data):
         png_pack(b'IDAT', zlib.compress(raw_data, 9)),
         png_pack(b'IEND', b'')])
  
+def geodetic_to_mercator(geodetic):
+    """This function takes an 2D array in geodetic coordinates (ie: lat x
+    lon unprojected) and converts it to web mercator.  This is needed
+    to correctly overlay an image on a leaflet map, which uses web
+    mercator by default.
+
+    This code works with arrays that match the relative proportions of
+    latitude and longitude of the earth, meaning that they have twice
+    as many points in longitude as latitude (i.e.: (180, 360) for 1
+    degree resolution).
+
+    The code for this is from:
+    http://stackoverflow.com/questions/25058880/convert-to-web-mercator-with-numpy
+
+    Parameters
+    ----------
+    geodetic: 2D numpy array
+        Latitude x Longitude 2D array
+
+    meractor: 2d numpy array
+
+    """
+    
+    def row2lat(row):
```

Comment:
Thanks for the comments!  Also, I think there is a small bug someplace, as
when I try to use this with actual data I get some bounds issues.  I'm
working on tracking it now now.  When I use a version of the mercator
projection code from elsewhere it works.

Do you want me to leave parameters/returns in `geodetic_to_mercator()` or
not?  I vote to have it there for consistency.  All other comments I have
no problem with at all.

On Mon, Aug 10, 2015 at 4:07 PM Filipe <notifications@github.com> wrote:

> In folium/utilities.py
> <https://github.com/python-visualization/folium/pull/168#discussion_r36677709>
> :
>
> > +    as many points in longitude as latitude (i.e.: (180, 360) for 1
> > +    degree resolution).
> > +
> > +    The code for this is from:
> > +    http://stackoverflow.com/questions/25058880/convert-to-web-mercator-with-numpy
>
>
> > +
> > +    Parameters
> > +    ----------
> > +    geodetic: 2D numpy array
> > +        Latitude x Longitude 2D array
> > +
> > +    meractor: 2d numpy array
> > +
> > +    """
> > +
> > +    def row2lat(row):
>
> Move this outside and name it _row2lat.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/python-visualization/folium/pull/168/files#r36677709>.
>